release/Earthfile

```diff
@@ -99,102 +186,55 @@ release-homebrew:
     RUN test -n "$RELEASE_TAG" && \
         test -n "$GIT_USERNAME" && \
         test -n "$GIT_NAME" && \
         test -n "$GIT_EMAIL" && \
         test -n "$HOMEBREW_EARTHLY_URL"
-    ARG NEW_URL=https://github.com/earthly/earthly/archive/"$RELEASE_TAG".tar.gz
+    ARG NEW_URL=https://github.com/"$GITHUB_USER"/"$EARTHLY_REPO"/archive/"$RELEASE_TAG".tar.gz
     RUN test -n "$NEW_URL"
-    GIT CLONE --branch main "$HOMEBREW_EARTHLY_URL" /earthly/homebrew-earthly
     WORKDIR /earthly/homebrew-earthly
 
-    # Git setup.
-    # Force use of https.
-    RUN git remote set-url origin "$HOMEBREW_EARTHLY_URL" && \
-        git remote -v
-    RUN --secret GIT_PASSWORD=+secrets/earthly-technologies/github/griswoldthecat/token \
-        git fetch --unshallow origin main
-    RUN git branch -f main && \
-        git checkout main && \
-        git branch -u origin/main
     RUN git config --global user.name "$GIT_NAME" && \
         git config --global user.email "$GIT_EMAIL"
+
     COPY ./envcredhelper.sh /usr/bin/envcredhelper.sh
     RUN git config --global credential.helper "/bin/sh /usr/bin/envcredhelper.sh"
 
+    RUN --secret GIT_PASSWORD="$GITHUB_TOKEN_SECRET_PATH" --no-cache \
+        git clone "https://$GITHUB_USER@github.com/$GITHUB_USER/$BREW_REPO.git" .
+
+    ARG RELEASE_BRANCH="release-$RELEASE_TAG"
+    RUN git switch -c "$RELEASE_BRANCH"
+
     # Make the change in a new branch.
-    RUN git checkout -b "release-$RELEASE_TAG"
+    #RUN git checkout -b "release-$RELEASE_TAG"
     RUN mkdir -p /params
     RUN curl -L "$NEW_URL" | sha256sum | cut -f 1 -d ' ' > /params/downloadsha256
     COPY --build-arg VERSION=$RELEASE_TAG \
         ../+earthly-darwin-amd64/tags ../+earthly-darwin-amd64/ldflags /params/
 
-    # Split /params/ldflags over two lines (to satisfy ruby linter).
-    RUN split --numeric-suffixes=1 --suffix-length=1 --bytes=80 /params/ldflags /params/ldflags && \
-        touch /params/ldflags2 && \
-        if [ -f "/params/ldflags3" ]; then \
-            echo "More than two lines for ldflags not supported" && \
-            exit 1 ;\
-        fi
-
     # replace version with #{version} variable to conform to homebrew PR requests
     RUN escapedversion=`echo "${RELEASE_TAG}" | sed 's/\./\\\./g'`; \
         sed -i -e "s/${escapedversion}/v#{version}/g" /params/ldflags*
 
     RUN sed -i \
-        -e 's^url ".*"^url "'"$NEW_URL"'"^' \
-        -e 's^sha256 ".*"$^sha256 "'$(cat /params/downloadsha256)'"^' \
-        -e 's^tags = ".*"^tags = "'"$(cat /params/tags)"'"^' \
-        -e 's^ldflags = ".*"^ldflags = "'"$(cat /params/ldflags1)"'"^' \
-        -e '/ldflags = ".*"/{n;s^".*"^"'"$(cat /params/ldflags2)"'"^}' \
+        -e 's^\burl ".*"^url "'"$NEW_URL"'"^' \
+        -e 's^\bsha256 ".*"$^sha256 "'$(cat /params/downloadsha256)'"^' \
+        -e 's^\btags = ".*"^tags = "'"$(cat /params/tags)"'"^' \
+        -e 's^\bldflags = ".*"^ldflags = "'"$(cat /params/ldflags)"'"^' \
```

Comment:
we no longer have to worry about the homebrew checks telling us this line is too long :tada: